### ISSUE TITLE
Calling move(...) on KinematicBody crashes game

### ISSUE THREAD

**Issue by haselkern** [okamstudio/godot#1284]
[my project file](http://www.mediafire.com/download/vve1kwi430guiuv/KinematicTest.zip)  
So, I was working on a small project when I noticed this crash. I created a simple project, and indeed calling move(...) on a KinematicBody crashes the game *without any information why that just happened*. I built Godot from source and this error persists.  
This is the script that I have attached to a KinematicBody:  
```
extends KinematicBody

func _ready():
	set_fixed_process(true)
	
func _fixed_process(delta):
	#This line crashes the game:
	move(Vector3(0, -1, 0))
	pass

```
This is the terminal output during execution (I added some comments starting with `#`)
```
#Starting godot (everything fine)
opening pack /mnt/data/Programming/GitHub/godot/bin/godot.x11.tools.32, -1, -1
opening pack /mnt/data/Programming/GitHub/godot/bin/data.pck, -1, -1
ERROR: _decode_variant: Condition ' params.size()!=5 ' is true. returned: Image()
   At: core/io/config_file.cpp:542.
ERROR: _load_recent_projects: Condition ' err!=OK ' is true. Continuing..:
   At: tools/editor/project_manager.cpp:523.
ERROR: _load_recent_projects: Condition ' err!=OK ' is true. Continuing..:
   At: tools/editor/project_manager.cpp:523.
#Opening project
OPENING: /mnt/data/Programming/Godot/KinematicTest (::mnt::data::Programming::Godot::KinematicTest)
#Opening scene
7
3
1
1
5
7
7
6
2
2
3
7
7
5
4
4
6
7
2
6
4
4
0
2
4
5
1
1
0
4
1
3
2
2
0
1
ERROR: fixed_material_set_parameter: fmp_emission expects Color
   At: servers/visual/rasterizer.cpp:392.
attempt set state: (current:0), (sources:res://test.gd)
#Starting game (while printing the following, the game crashes)
SAVING: res://new_scene.scn
UPDATING: res://new_scene.scn
UPDATING: res://test.gd
UPDATING: res://test.gd
running: /mnt/data/Programming/GitHub/godot/bin/godot.x11.tools.32 -path /mnt/data/Programming/Godot/KinematicTest -rdebug localhost:6007 res://new_scene.scn
ERROR: set_stream_peer: Condition ' p_peer.is_null() ' is true.
   At: core/io/packet_peer.cpp:228.
********* listening on port 6007
OK! 8
opening pack /mnt/data/Programming/GitHub/godot/bin/godot.x11.tools.32, -1, -1
opening pack /mnt/data/Programming/Godot/KinematicTest/data.pck, -1, -1
has connection!
has connection!
7
3
1
1
5
7
7
6
2
2
3
7
7
5
4
4
6
7
2
6
4
4
0
2
4
5
1
1
0
4
1
3
2
2
0
1
ERROR: fixed_material_set_parameter: fmp_emission expects Color
   At: servers/visual/rasterizer.cpp:392.
CLOSING CONNECTION
ERROR: set_stream_peer: Condition ' p_peer.is_null() ' is true.
   At: core/io/packet_peer.cpp:228.
#Closing Godot
freed code id 5
freed code id 4
freed code id 3
ERROR: free: Method/Function Failed.
   At: servers/visual/visual_server_raster.cpp:4118.
freed code id 6
ERROR: free: Method/Function Failed.
   At: servers/visual/visual_server_raster.cpp:4118.
ERROR: free: Method/Function Failed.
   At: servers/visual/visual_server_raster.cpp:4118.
ERROR: set_stream_peer: Condition ' p_peer.is_null() ' is true.
   At: core/io/packet_peer.cpp:228.
base resources missing 4
ERROR: finish: Condition ' instance_dependency_map.size() ' is true.
   At: servers/visual/visual_server_raster.cpp:6757.
```
I'm currently using Ubuntu 14.04 32bit and `uname -a` returns this (if that helps):
```
Linux lars-desktop 3.13.0-44-generic #73-Ubuntu SMP Tue Dec 16 00:23:46 UTC 2014 i686 i686 i686 GNU/Linux
```

**Comment by reduz** [comment]
despite the errors, it seems it just exited normally

**Comment by UsernameIsAReservedWord** [comment]
erros mentionned are not related.

I have `move()` that make crashes Windows 32bits versions of the engine.

The same version compiled in 64bits does not crash.

**Comment by UsernameIsAReservedWord** [comment]
I managed to test it with the Visual Studio 2010 debugger.

It crashes into `collision_solver_sw.cpp`, line 299, into the `CollisionSolverSW::solve_distance_plane()`.

My test-scene contains a `StaticBody` with a `Plane` collision shape as floor, and a simple `KinematicBody` with a sphere as collision shape.
The script attached to the KinematicBody is the same as @haselkern posted.

**Comment by reduz** [comment]
oh is there a shape using plane collision in the scene?

**Comment by UsernameIsAReservedWord** [comment]
this line : `supports[i] = p_transform_B.xform( supports[i] );`

the debugger says : 

closest_d = -2.3215289e+032
i = 1255
support_count = 2048

but `supports[]` seems to contain only 16 Vector3.

which means buffer overflow

````
	for(int i=0;i<support_count;i++) {

		supports[i] = p_transform_B.xform( supports[i] ); 
		real_t d = p.distance_to(supports[i]);
		if (i==0 || d<closest_d) {
			closest=supports[i];
			closest_d=d;
			if (d<=0)
				collided=true;
		}

	}
````

**Comment by UsernameIsAReservedWord** [comment]
@reduz  : yes, there is a shape using a Plane.

**Comment by UsernameIsAReservedWord** [comment]
I've published a fix for this : #1446
